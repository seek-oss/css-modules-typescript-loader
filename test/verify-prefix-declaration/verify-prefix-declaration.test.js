const fs = require('fs');
const compiler = require('../compiler.js');

test('Should have prefixed interface name', async () => {
  expect.assertions(1);

  await compiler(require.resolve('./index.js'), {
    prefix: 'I'
  });

  const declaration = fs.readFileSync(
    require.resolve('./index.css.d.ts'),
    'utf-8'
  );

  expect(declaration).toMatchSnapshot();
});
