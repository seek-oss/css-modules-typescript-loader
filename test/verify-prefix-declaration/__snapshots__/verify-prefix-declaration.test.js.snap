// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should have prefixed interface name 1`] = `
"// This file is automatically generated.
// Please do not change this file!
interface ICssExports {
  'hyphened-classname': string;
  'otherClass': string;
  'someClass': string;
  'underscored_classname': string;
  'validClass': string;
}
declare const cssExports: ICssExports;
export = cssExports;
"
`;
